jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/mapping/TestUnidirectionalOneToMany.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     09/12/2018 - Will Dazey
+//       - 391279: Add support for Unidirectional OneToMany mappings with non-nullable values
+package org.eclipse.persistence.jpa.test.mapping;
+
+import java.util.ArrayList;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+import org.eclipse.persistence.jpa.test.framework.DDLGen;
+import org.eclipse.persistence.jpa.test.framework.Emf;
+import org.eclipse.persistence.jpa.test.framework.EmfRunner;
+import org.eclipse.persistence.jpa.test.framework.Property;
+import org.eclipse.persistence.jpa.test.mapping.model.CommentA;
+import org.eclipse.persistence.jpa.test.mapping.model.CommentB;
+import org.eclipse.persistence.jpa.test.mapping.model.ComplexIdA;
+import org.eclipse.persistence.jpa.test.mapping.model.ComplexIdB;
+import org.eclipse.persistence.jpa.test.mapping.model.PostA;
+import org.eclipse.persistence.jpa.test.mapping.model.PostB;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(EmfRunner.class)
+public class TestUnidirectionalOneToMany {
+    @Emf(createTables = DDLGen.DROP_CREATE, 
+            classes = { CommentA.class, CommentB.class, ComplexIdA.class, ComplexIdB.class, PostA.class, PostB.class }, 
+            properties = { @Property(name = "eclipselink.cache.shared.default", value = "false") })
```

Comment:
when I add

```
@Property(name = "eclipselink.logging.parameters", value = "true"),
@Property(name = "eclipselink.logging.level.sql", value = "FINEST")
```

here, in the log, there I can see:
```
[EL Fine]: sql: 2018-09-17 13:54:42.12--ClientSession(1835848160)--Connection(460674260)--INSERT INTO COMMENTA (ID, POSTA_FK) VALUES (?, ?)
	bind => [4, 9]
[EL Fine]: sql: 2018-09-17 13:54:42.121--ClientSession(1835848160)--Connection(460674260)--UPDATE COMMENTA SET POSTA_FK = ? WHERE (ID = ?)
	bind => [9, 4]
```

I'd expect, that after this patch, the update statement won't be executed

Reply:
Ya, I'm having troubles with the 'postInsert' event. See, what happens is that 'postInsert' for Post, we create a 'PostInsert' event, but it's just meant for the Post object being inserted! For 'PostUpdate' we have one event PER Comment object, so it's pretty easy to remove from the List of data modification events. 
If I remove the whole 'PostInsert' event for the first Comment, then the rest will not know there was an event! Causing the rest of the INSERT COMMENT to fail.

Reply:
Ok @lukasj , I think I fixed this issue by reworking how 'PostInsert' works. I changed it now so that an individual 'PostInsert' event gets created for EACH Comment object. The current way it works is that only 1 'PostInsert' event gets created and during execution (still on transaction commit), we loop through all the relationship objects and UPDATE each one. 
Now, we create an event for each relationship object and each event just runs for that one object. I think this basically has the same logic, but I'd appreciate an objective look and thoughts/tests I don't have.